### DISCUSSION TITLE
Sentence trigger API will be changing

### DISCUSSION THREAD

**Discussion by synesthesiam** [zachowj/hass-node-red#323] [Feature Requests]
Just a heads up that the function signature for sentence trigger handling will be changing again soon: https://github.com/home-assistant/core/pull/131982

**Comment by zachowj** [comment]
thanks for the heads-up

**Comment by zachowj** [comment]
After reviewing the PR, it doesn't appear to directly affect our use case. However, the issue seems to be related to the `hassil` library and the [RecognizeResult](https://github.com/home-assistant/hassil/blob/3bf179c0e840460692406480bcd661dcbed7edc9/hassil/recognize.py#L28) class. It looks like changes were made to this class, causing it to no longer be serializable by the default JSON encoder.

**Comment by synesthesiam** [comment]
Setting the `intent_sentence` field to `None` before serializing may help.
